test/partial-result-stream.ts

```diff
@@ -309,6 +370,50 @@ describe('PartialResultStream', () => {
         );
     });
 
+    it('should emit non-retryable error', done => {
+      // This test will emit two rows and then an error.
+      const fakeCheckpointStream = through.obj();
+      // tslint:disable-next-line no-any
+      const resetStub = ((fakeCheckpointStream as any).reset = () => {});
+      sandbox.stub(checkpointStream, 'obj').returns(fakeCheckpointStream);
+
+      const fakeRequestStream = through.obj();
+
+      const requestFnStub = sandbox.stub();
+
+      requestFnStub.onCall(0).callsFake(() => {
+        setTimeout(() => {
+          fakeRequestStream.push(RESULT_WITH_TOKEN);
+          fakeCheckpointStream.emit('checkpoint', RESULT_WITH_TOKEN);
+          fakeRequestStream.push(RESULT_WITH_TOKEN);
+          fakeCheckpointStream.emit('checkpoint', RESULT_WITH_TOKEN);
+
+          setTimeout(() => {
+            fakeRequestStream.emit('error', {
+              code: status.DATA_LOSS,
+              message: 'Non-retryable error.',
+            } as ServiceError);
+            fakeRequestStream.end();
```

Comment:
Same above. It can be deleted.

Reply:
`end()` call removed from test case.

In case of a non-retryable error, we circumvent the `requestsStream` and send an error directly to the stream that is returned to the user (technically, by sending it first to the `batchAndSplitOnTokenStream`). So it is not necessary to end the `requestsStream` in that case.

See https://github.com/googleapis/nodejs-spanner/blob/569570fe3ed7a8ebc10b0ed2d528435102be5e82/src/partial-result-stream.ts#L383